Description: Use database contrib_regression rather than postgres
Forwarded: no
Author: Markus Wanner <markus.wanner@datahouse.ch>
Last-Update: 2015-06-18
--- a/sql/mysql_fdw.sql
+++ b/sql/mysql_fdw.sql
@@ -1,4 +1,4 @@
-\c postgres postgres
+\c contrib_regression postgres
 CREATE EXTENSION mysql_fdw;
 CREATE SERVER mysql_svr FOREIGN DATA WRAPPER mysql_fdw;
 CREATE USER MAPPING FOR postgres SERVER mysql_svr OPTIONS(username 'foo', password 'bar');
--- a/expected/mysql_fdw.out
+++ b/expected/mysql_fdw.out
@@ -1,4 +1,4 @@
-\c postgres postgres
+\c contrib_regression postgres
 CREATE EXTENSION mysql_fdw;
 CREATE SERVER mysql_svr FOREIGN DATA WRAPPER mysql_fdw;
 CREATE USER MAPPING FOR postgres SERVER mysql_svr OPTIONS(username 'foo', password 'bar');
